### PR TITLE
STB-2824 - Implement minimum search length validation across user and firm search endpoints

### PR DESCRIPTION
**Pull request by snowy-snowdon**
This pull request updates the search validation logic for firm search endpoints in the `UserController`, ensuring that only empty or whitespace-only queries are rejected and that single-character queries are now processed. It also adds comprehensive unit tests to verify this behavior and refactors some import statements for clarity.

Search validation improvements:

* Updated the `getFirms` and `searchFirms` methods in `UserController` so that only empty or whitespace-only queries are rejected, allowing single-character queries to be processed. This prevents unnecessary service calls for blank queries and supports 1-character search terms. [[1]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R111-R115) [[2]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R476-R491)
* Added logic to trim queries in `searchFirms` before passing them to the service, ensuring consistency in query handling.

Unit test additions and improvements:

* Added three new test classes (`SearchValidationMinimumOneTest`, `SearchValidationTest`, and `SearchValidationUnitTest`) to verify that the endpoints correctly handle empty, whitespace, and single-character queries, and that service methods are only called for valid queries. These tests cover both normal and edge cases. [[1]](diffhunk://#diff-b88560de0b3ad1b3246251525c5e14fdb8dc1ae67d6200f01eeb660295672c00R1-R95) [[2]](diffhunk://#diff-4128716dde48d77da8bfaba8d4dec5a9857a776451fbe56c146b3d9f4ea9b4c3R1-R165) [[3]](diffhunk://#diff-9ee550cf0bec9206c63568f80b6ff1a4c07e5c26259948929d9c7f5d42720305R1-R138)

Code cleanup and refactoring:

* Refactored import statements in `UserController` for better organization and clarity, including moving static imports and adjusting the location of some imports. [[1]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L3-L5) [[2]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L20) [[3]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R35) [[4]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R78-R79)
* Updated imports and static imports in `UserControllerTest` to improve readability and maintain consistency with the new validation logic.

### CODE REVIEW

**Review by pliao-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/justice/laa/portal/landingpage/controller/UserController.java [hunk 1]
```diff
@@ -471,8 +475,15 @@ public String createUserFirm(FirmSearchForm firmSearchForm, HttpSession session,
...
                                                  @RequestParam(value = "firmSearchResultCount", defaultValue = "10") Integer count) {
+        // If the query is blank/whitespace-only, return an empty result and do not
+        // call the service to avoid unnecessary work.
+        if (query == null || query.trim().isEmpty()) {
+            return new ArrayList<>();
+        }
+
         int validatedCount = Math.max(10, Math.min(count, 100));
-        List<FirmDto> firms = firmService.searchFirms(query);
+        String trimmedQuery = query.trim();
```
[the remark this answers is not in the document]
Comment (snowy-snowdon) [reply src/main/java/uk/gov/justice/laa/portal/landingpage/controller/UserController.java]: Updated to remove additonal variable. Now doing this

`List<FirmDto> firms = firmService.searchFirms(query.trim());`

**Review by sureshkavali-moj — APPROVED** [review]
